### PR TITLE
FTW Bakeoff losses ablation

### PR DESCRIPTION
**Pull request by Gedeon-m-gedus**
## Added Losses Ablation
Refactored and extended the loss configuration for bakeoff loss ablation.

### Changes
- **Refactored** all custom and standard loss functions into `ftw_tools/training/losses.py`
- **Updated** `CustomSemanticSegmentationTask` to import from the centralized `losses.py`
- Supported loss types include:
  - `ce` — Cross Entropy
  - `pixel_weighted_ce` — Gaussian-blurred, per-pixel weighted CE
  - `jaccard` — IoU-based Jaccard Loss
  - `focal` — Focal Loss
  - `dice` — Dice Loss
  - `ce+dice` — Combined Cross Entropy + Dice
  - `logcoshdice` — Log-Cosh smoothed Dice
  - `logcoshdice+ce` — Log-Cosh smoothed Dice + Cross Entropy
  - `ftnmt` — Fractal Tanimoto (FTNMT) Loss
  - `ce+ftnmt` — Combined CE + FTNMT
  - `localtversky` — Locally Weighted Tversky Focal Loss
  - `tversky_ce` — Tversky + Cross Entropy combined loss

### CODE REVIEW

**Comment by calebrob6** [comment]
@Gedeon-m-gedus there is inconsistent handling of the ignore_index among these -- zero'ing out the ignored pixels is not the same as ignoring in the loss.
